skribenten-backend/src/main/kotlin/no/nav/pensjon/brev/skribenten/routes/SakBrev.kt

```diff
@@ -0,0 +1,138 @@
+import no.nav.pensjon.brev.skribenten.services.BrevredigeringService
+import no.nav.pensjon.brev.skribenten.services.PenService
+import java.time.LocalDateTime
+
+fun Route.sakBrev(brevredigeringService: BrevredigeringService) =
+    route("/brev") {
+        post<OpprettBrevRequest> { request ->
+            val sak: PenService.SakSelection = call.attributes[AuthorizeAnsattSakTilgang.sakKey]
+
+            brevredigeringService.opprettBrev(call, sak, request.brevkode, request.saksbehandlerValg) {
+                OppprettBrevResponse(
+                    id = id.value,
+                    redigertBrev = redigertBrev,
+                    sistredigert = sistredigert,
+                    brevkode = Brevkode.Redigerbar.valueOf(brevkode),
+                    saksbehandlerValg = saksbehandlerValg,
+                )
+            }.map { brev ->
+                call.respond(HttpStatusCode.OK, brev)
+            }.catch { message, statusCode ->
+                call.application.log.error("$statusCode - Feil ved oppretting av brev ${request.brevkode}: $message")
+                call.respond(HttpStatusCode.InternalServerError, "Feil ved oppretting av brev.")
+            }
+        }
+
+        post<OppdaterBrevRequest>("/{brevId}") { request ->
+            val brevId = call.parameters.getOrFail<Long>("brevId")
+            brevredigeringService.oppdaterBrev(call, brevId, request.brevkode, request.saksbehandlerValg, request.redigertBrev) {
+                OppprettBrevResponse(
+                    id = brevId,
+                    redigertBrev = redigertBrev,
+                    sistredigert = sistredigert,
+                    brevkode = request.brevkode,
+                    saksbehandlerValg = saksbehandlerValg
+                )
+            }.map { brev ->
+                call.respond(HttpStatusCode.OK, brev!!)
+            }.catch { message, statusCode ->
+                call.application.log.error("$statusCode - Feil ved oppdatering av brev ${request.brevkode}: $message")
+                call.respond(HttpStatusCode.InternalServerError, "Feil ved oppdatering av brev.")
+            }
+        }
+
+        delete("/{brevId}") {
+            val brevId = call.parameters.getOrFail<Long>("brevId")
+            brevredigeringService.slettBrev(brevId).map {
+                call.respond(HttpStatusCode.OK)
+            }.catch { message, httpStatusCode ->
+                call.application.log.error("$httpStatusCode - Feil ved sletting av brev med id: $brevId, $message")
+            }
+        }
+
+        get("/{brevId}") {
+            val brevId = call.parameters.getOrFail<Long>("brevId")
+            val response = brevredigeringService.hentBrev(brevId, mapper)
+
+            if (response != null) {
+                call.respond(HttpStatusCode.OK, response)
+            } else {
+                call.respond(HttpStatusCode.NotFound, "Brev not found")
+            }
+        }
+
+        get("/{navident}") {
```

Comment:
Navident bør ikke komme fra url, da vil vi antyde at vi godtar at man henter brev for andre enn seg selv. Navident finner du i principal.